src/Cli/dotnet/commands/dotnet-check-update/ProductCollectionProvider.cs

```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using Microsoft.Deployment.DotNet.Releases;
+
+namespace Microsoft.DotNet.Tools.CheckUpdate
+{
+    public class ProductCollectionProvider : IProductCollectionProvider
+    {
+        public ProductCollection GetProductCollection()
+        {
+            return ProductCollection.GetAsync().Result;
```

Comment:
If the service is hit really hard, it will respond with HTTP 503. I believe the library will just pass that exception to you. I'm take it the command will just end. What do we generally do with other commands that have external resource dependencies when they fail?

One option might be to cache the release.json and fallback to a local copy (aka offline mode), but that's probably another feature.

Reply:
> What do we generally do with other commands that have external resource dependencies when they fail?

I'm not sure, but I imagine we just surface that error to the user. 

> One option might be to cache the release.json and fallback to a local copy (aka offline mode), but that's probably another feature.

Agreed, this would be nice but that would be a separate change.